fdbserver/BlobMigrator.actor.cpp

```diff
@@ -573,6 +567,33 @@ class BlobMigrator : public NonCopyable, public ReferenceCounted<BlobMigrator> {
 		}
 	}
 
+	// This API is used by DD to figure out split points for data movement.
+	ACTOR static Future<Void> processSplitMetricsRequest(Reference<BlobMigrator> self, SplitMetricsRequest req) {
+		SplitMetricsReply rep;
+		int64_t bytes = 0; // number of bytes accumulated for current split
+		for (auto granule : self->blobGranules_) {
+			if (!req.keys.contains(granule.keyRange)) {
+				continue;
+			}
+			bytes += granule.sizeInBytes;
+			if (bytes < req.limits.bytes) {
+				continue;
+			}
+			// Add a split point if the key range exceeds expected minimal size in bytes
+			rep.splits.push_back_deep(rep.splits.arena(), granule.keyRange.end);
+			bytes = 0;
+			// Limit number of splits in single response for fast RPC processing
+			if (rep.splits.size() > SERVER_KNOBS->SPLIT_METRICS_MAX_ROWS) {
```

Comment:
We should add a code probe here to make sure this logic actually gets hit, as well as potentially in the native api for the logic to handle when the split points response is incomplete

Reply:
Actually it doesn't look like the more flag is used anywhere in the client? Is that intentional?

Reply:
added a code probe. Thanks!

The more flag is used in NativeAPI::splitStorageMetricsWithLocations(). It'll send another request if response has the more flag

```
     if (res.more && res.splits.size()) {
	// Next request will return split points after this one
	beginKey = KeyRef(beginKey.arena(), res.splits.back());
    } else {
	break;
   }
```